utils_linux.go

```diff
@@ -28,6 +29,48 @@ import (
 
 var errEmptyID = errors.New("container id cannot be empty")
 
+var cliSystemdProperties = map[string]struct {
+	Name string // systemd property name
+	conv func(name, val string) (interface{}, error)
+}{
+	"stop-timeout": {"TimeoutStopUSec", convertUSec},
```

Comment:
We don't want to explicitly add each property. We want to support arbitrary system properties. For e.g. `--systemd-property TimeoutStopUSec=1000`. This was requested on the original PR.

Reply:
Initially I also wanted to make it generic, but it seems it is both impossible and useless to support arbitrary systemd properties. Let me explain.

It is impossible because we do not know types of arguments. I know that systemd dbus uses at least booleans, strings, various types of integers, and even arrays. I don't see a way to pass the argument (to systemd dbus) without knowing its type.

It is useless because a limited amount of properties are supported for `StartTransientUnit()` systemd scope. Here's a quote from the docs (that might be obsoleted, but still):

> The properties array of StartTransientUnit() may take many of the settings that may also be configured in unit files. Not all parameters are currently accepted though, but we plan to cover more properties with future release. Currently you may set the Description, Slice and all dependency types of units, as well as RemainAfterExit, ExecStart for service units, TimeoutStopUSec and PIDs for scope units, and CPUAccounting, CPUShares, BlockIOAccounting, BlockIOWeight, BlockIOReadBandwidth, BlockIOWriteBandwidth, BlockIODeviceWeight, MemoryAccounting, MemoryLimit, DevicePolicy, DeviceAllow for services/scopes/slices. These fields map directly to their counterparts in unit files and as normal D-Bus object properties. The exception here is the PIDs field of scope units which is used for construction of the scope only and specifies the initial PIDs to add to the scope object.

From the above list, runc already sets `Slice`, `Description`, `DefaultDependencies`, and `PIDs`, as well as (I assume) all the cgroup parameters. Let's see what is left to set for a scope unit...

It seems the only property left is `TimeoutStopUSec`, there's no other option we can set.

Still, the doc can be obsoleted, or I might be wrong, so I modified the code in a way that allows us to add an option easily (in case of usec option that would be a line of code, in case of option with other type of arg you'd need to provide a validator+converter function, one for each type).

Reply:
> We don't want to explicitly add each property

chould it be an annotation?

These runc-only features make interoperability with other OCI runtimes more difficult.  If it is an annotation, other runtimes that have no support for it (including older runc versions) will ignore the timeout.

Reply:
OK the (seemingly up-to-date) list of properties available for transient units is here: https://github.com/systemd/systemd/blob/master/docs/TRANSIENT-SETTINGS.md